### PR TITLE
hotfix: bad big number behaviour

### PR DESCRIPTION
**Pull request by Brendan-Lucas**
Fix bad decimal behaviour associated with simulate and calculate in YTC

moved to using decimal library for math and translating from decimal to hex for use in eth big number to avoid usage of exponent values.

### CODE REVIEW

File: frontend/tailwind.config.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module.exports = {
-   purge: ['./src/**/*.{js,jsx,ts,tsx}', './public/index.html'],
-    darkMode: false, // or 'media' or 'class'
+    content: [
```
Comment (mattiaslightstone) [inline frontend/tailwind.config.js:2]: Why the tailwind changes?

File: frontend/src/api/prices/element.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export const isElementSpecificToken = (baseTokenNameLowercase: string) => (
+  ELEMENT_SPECIFIC_TOKENS.includes(baseTokenNameLowercase)
+)
+
+// written to use this contract: https://etherscan.io/address/0x557f483e13c94a02cbfefe1d27a781c4221f37f5#readContract
+export const getPriceValueFromContract = async (
+  contractAddress: string,
+): Promise<number> =>  {
+  console.log("getPriceValueFromContract")
```
Comment (mattiaslightstone) [inline frontend/src/api/prices/element.ts]: do we want a console log here?

File: frontend/src/api/prices/element.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  if (contractAddress) {
```
Comment (mattiaslightstone) [inline frontend/src/api/prices/element.ts]: if the type is string is this necessary? Are you expecting an empty string potentially?

**Review by mattiaslightstone — COMMENTED** [review]
Looks great, a few minor questions, but nothing blocking approval

File: frontend/src/api/prices/element.ts (hunk 2 above)
Comment (Brendan-Lucas) [reply frontend/src/api/prices/element.ts]: no we do not.

File: frontend/src/api/prices/element.ts (hunk 3 above)
Comment (Brendan-Lucas) [reply frontend/src/api/prices/element.ts]: fair point

File: frontend/tailwind.config.js (hunk 1 above)
Comment (Brendan-Lucas) [reply frontend/tailwind.config.js:2]: unrelated to fix but resolves some build errors as tailwind no longer uses purge as a library.

**Review by mattiaslightstone — APPROVED** [review]

**Pull request merged**